src/index/IndexImpl.cpp

```diff
@@ -966,112 +1050,74 @@ void IndexImpl::readIndexBuilderSettingsFromFile() {
                                          allIntegersBecomeDoubles};
-  std::string key = "parser-integer-overflow-behavior";
-  if (j.count(key)) {
-    auto value = j[key];
-    if (value == overflowingIntegersThrow) {
-      LOG(INFO) << "Integers that cannot be represented by QLever will throw "
-                   "an exception"
-                << std::endl;
-      turtleParserIntegerOverflowBehavior_ =
-          TurtleParserIntegerOverflowBehavior::Error;
-    } else if (value == overflowingIntegersBecomeDoubles) {
-      LOG(INFO) << "Integers that cannot be represented by QLever will be "
-                   "converted to doubles"
-                << std::endl;
-      turtleParserIntegerOverflowBehavior_ =
-          TurtleParserIntegerOverflowBehavior::OverflowingToDouble;
-    } else if (value == allIntegersBecomeDoubles) {
-      LOG(INFO) << "All integers will be converted to doubles" << std::endl;
-      turtleParserIntegerOverflowBehavior_ =
-          TurtleParserIntegerOverflowBehavior::OverflowingToDouble;
-    } else {
-      AD_CONTRACT_CHECK(std::find(allModes.begin(), allModes.end(), value) ==
-                        allModes.end());
-      LOG(ERROR) << "Invalid value for " << key << std::endl;
-      LOG(INFO) << "The currently supported values are "
-                << absl::StrJoin(allModes, ",") << std::endl;
-    }
-  } else {
+  if (parserIntegerOverflowBehavior == overflowingIntegersThrow) {
+    LOG(INFO) << "Integers that cannot be represented by QLever will throw "
+                 "an exception"
+              << std::endl;
     turtleParserIntegerOverflowBehavior_ =
         TurtleParserIntegerOverflowBehavior::Error;
-    LOG(INFO) << "Integers that cannot be represented by QLever will throw an "
-                 "exception (this is the default behavior)"
+  } else if (parserIntegerOverflowBehavior ==
+             overflowingIntegersBecomeDoubles) {
+    LOG(INFO) << "Integers that cannot be represented by QLever will be "
+                 "converted to doubles"
               << std::endl;
+    turtleParserIntegerOverflowBehavior_ =
+        TurtleParserIntegerOverflowBehavior::OverflowingToDouble;
+  } else if (parserIntegerOverflowBehavior == allIntegersBecomeDoubles) {
+    LOG(INFO) << "All integers will be converted to doubles" << std::endl;
+    turtleParserIntegerOverflowBehavior_ =
+        TurtleParserIntegerOverflowBehavior::OverflowingToDouble;
+  } else {
+    // TODO This can maybe be replaced with a validator, if the logging of the
+    // information is not needed.
+    AD_CONTRACT_CHECK(std::find(allModes.begin(), allModes.end(),
+                                parserIntegerOverflowBehavior) ==
+                      allModes.end());
+    LOG(ERROR) << "Invalid value for parser-integer-overflow-behavior"
+               << std::endl;
+    LOG(INFO) << "The currently supported values are "
+              << absl::StrJoin(allModes, ",") << std::endl;
```

Comment:
I like the idea of a validator.
But what I like even more is to handle the conversion from the String to the enum directly in the config manager, s.t. we can for example call
````
configManager.addOptionWithConversion<string, someEnum>(path, description, &enum, possiblyThrowingConverterFunctionFromStringToEnum).


````

You can maybe already think of a way to incorporate this into the configManager.

Reply:
Hmm, that is a tough one. However, I'm not so sure, if extending the `addOption` family would be the best idea in terms of clarity. Having 8 options for adding an option, with half being radically different to the other half, does sound a bit confusing.

I'm also not sure, if we can get around the need for explicitly stating the needed types in `<>`. It would be kinda weird, to have to pass variables for intermediate results.

My first thought, on how to accomplish this while converter idea, would have been to use the same style as for `addValidator`, but with having to pass a variable address at the beginning, to put the result in.
Of course, that would still require the user to personally build the variables for the intermediate results, so I'm not the biggest fan.